client/src/components/Voting/Voting.js

```diff
@@ -21,20 +23,90 @@ const Voting = () => {
         isCanceled,
       })
       setMovies([])
+      setVoting([])
+      setIsLoadingUser(true)
+      setIsLoading(true)
+    } catch (error) {
+      console.error(error)
+      setComponentError({ message: error.message })
+    }
+  }
+
+  const unknownMovieHandler = async direction => {
+    let unknownMovies = {}
+    switch (direction) {
+      case 'UP':
+        unknownMovies = { firstMovie: movies[0]._id, secondMovie: '' }
+        break
+      case 'DOWN':
+        unknownMovies = { firstMovie: movies[1]._id, secondMovie: '' }
+        break
+      default:
+        unknownMovies = {
+          firstMovie: movies[0]._id,
+          secondMovie: movies[1]._id,
+        }
+    }
+
+    try {
+      await axios.patch(`/api/users/unknownmovies/${userData._id}`, {
+        firstMovie: unknownMovies.firstMovie,
+        secondMovie: unknownMovies.secondMovie,
+      })
+      await axios.patch(`/api/votings/${voting._id}`, {
+        firstMovieWon: false,
+        isCanceled: true,
+      })
+      setMovies([])
+      setVoting([])
+      setIsLoadingUser(true)
       setIsLoading(true)
     } catch (error) {
       console.error(error)
       setComponentError({ message: error.message })
     }
   }
 
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const token = JSON.parse(localStorage.getItem('authToken'))
+
+        if (token === '') {
+          throw new Error({ message: 'Not logged in' })
+        }
+
+        const config = {
+          headers: {
+            'x-auth-token': token,
+          },
+        }
+
+        const user = await axios.get('/api/auth', config)
+        setUserData(user.data)
+        setIsLoadingUser(false)
+      } catch (error) {
+        console.error(error)
+        setComponentError({ message: error.message })
+        setIsLoadingUser(false)
+        onLogout()
+      }
+    }
+
+    if (isLoadingUser) {
+      fetchUser()
+    }
+  }, [isLoadingUser, onLogout])
```

Comment:
`onLogout` is changed on every render of your `App`. This would trigger this `useEffect` which might lead to more issues.
You could use a `useCallback`:
https://reactjs.org/docs/hooks-reference.html#usecallback